src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -188,6 +188,54 @@ public function getBoolean($key, $default = false)
     }
 
     /**
+     * Returns the parameter value converted to a DateTime object.
+     *
+     * @param string             $key      The parameter key
+     * @param string             $format   The expected date format
+     * @param \DateTime|null     $default  The default value if the parameter key does not exist
+     * @param \DateTimeZone|null $timeZone
+     *
+     * @return \DateTime|null
+     */
+    public function getDate($key, $format = 'Y-m-d', $default = null, $timeZone = null)
+    {
+        if (!$this->has($key)) {
+            return $default;
```

Comment:
Could very well be, I like this more than manually constructing a DateTime object for the default parameter.

So you can call it as follows: `$bag->getDate('from', 'Y-m-d', '2016-01-01')`

Yeah, perhaps we can go this way.